migrations/1630928992986-addBucketToFiles.js

```diff
@@ -0,0 +1,77 @@
+const mongoose = require('mongoose');
+// eslint-disable-next-line no-unused-vars
+const { alert, error } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+const { Schema } = mongoose;
+
+const FileModel = mongoose.model(
+	'file_20210906',
+	new mongoose.Schema({
+		refOwnerModel: { type: String },
+		owner: { type: Schema.Types.ObjectId },
+		bucket: { type: String },
+	}),
+	'files'
+);
+
+const UserModel = mongoose.model(
+	'user_20210906',
+	new mongoose.Schema({
+		schoolId: { type: Schema.Types.ObjectId },
+	}),
+	'users'
+);
+
+const CourseModel = mongoose.model(
+	'course_20210906',
+	new mongoose.Schema({
+		schoolId: { type: Schema.Types.ObjectId },
+	}),
+	'courses'
+);
+
+const TeamModel = mongoose.model(
+	'team_20210906',
+	new mongoose.Schema({
+		schoolIds: { type: [Schema.Types.ObjectId] },
+	}),
+	'teams'
+);
+
+const ownerModel = {
+	user: UserModel,
+	course: CourseModel,
+	teams: TeamModel,
+};
+
+module.exports = {
+	up: async function up() {
+		alert('Start adding buckets to file documents...');
+		await connect();
+		const files = FileModel.find({ bucket: { $exists: false } });
```

Comment:
lean().exec() ? we might need to paginate this (a lot of files might exist here... maybe use chunks of 100 files...

Reply:
`lean` is not possible in this case, because we want to modify the files with `file.save()`.
And `exec` is not needed, because the Mongoose query can already work as async iterator.